monkey/monkey_island/cc/services/authentication_service/register_event_handlers.py

```diff
@@ -0,0 +1,28 @@
+from common import DIContainer
+from common.agent_events import AbstractAgentEvent, AgentShutdownEvent
+from common.event_queue import IAgentEventQueue
+from common.types import AgentID
+from monkey_island.cc.event_queue import IIslandEventQueue, IslandEventTopic
+
+from .authentication_facade import AuthenticationFacade
+
+
+def register_event_handlers(container: DIContainer, authentication_facade: AuthenticationFacade):
+    agent_event_queue = container.resolve(IAgentEventQueue)
+    island_event_queue = container.resolve(IIslandEventQueue)
+    agent_remover = AgentRemover(authentication_facade)
+
+    agent_event_queue.subscribe_type(AgentShutdownEvent, agent_remover.remove_on_shutdown)
+    island_event_queue.subscribe(IslandEventTopic.AGENT_TIMED_OUT, agent_remover.remove_on_timeout)
+
+
+class AgentRemover:
```

Comment:
`AgentUserRemover`. This name makes it seem like we're removing agents from the AgentRepository.